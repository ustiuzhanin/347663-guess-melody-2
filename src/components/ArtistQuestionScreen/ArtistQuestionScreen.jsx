import React, {PureComponent} from 'react';
import AudioPlayer from '../AudioPlayer/AudioPlayer.jsx';
import PropTypes from 'prop-types';

export default class ArtistQuestionScreen extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isPlaying: false
    };
  }

  render() {
    const {screenIndex, question, onAnswer} = this.props;
    const {answers, song} = question;
    return (
      <section className='game game--artist'>
        <header className='game__header'>
          <a className='game__back' href='#'>
            <span className='visually-hidden'>Сыграть ещё раз</span>
            <img
              className='game__logo'
              src='img/melody-logo-ginger.png'
              alt='Угадай мелодию'
            />
          </a>

          <svg
            xmlns='http://www.w3.org/2000/svg'
            className='timer'
            viewBox='0 0 780 780'
          >
            <circle
              className='timer__line'
              cx='390'
              cy='390'
              r='370'
              style={{
                filter: `url(#blur)`,
                transform: `rotate(-90deg) scaleY(-1)`,
                transformOrigin: `center`
              }}
            />
          </svg>

          <div className='timer__value' xmlns='http://www.w3.org/1999/xhtml'>
            <span className='timer__mins'>05</span>
            <span className='timer__dots'>:</span>
            <span className='timer__secs'>00</span>
          </div>

          <div className='game__mistakes'>
            <div className='wrong'></div>
            <div className='wrong'></div>
            <div className='wrong'></div>
          </div>
        </header>

        <section className='game__screen'>
          <h2 className='game__title'>Кто исполняет эту песню?</h2>
          <div className='game__track'>
            <div className='track'>
              <AudioPlayer
                src={song.src}
                isPlaying={this.state.isPlaying}
                onPlayButtonClick={() =>
                  this.setState({isPlaying: !this.state.isPlaying})
                }
              />
            </div>
          </div>

          <form className='game__artist'>
            {answers.map((answer, i) => {
              return (
                <div key={`${screenIndex}${i}`} className='artist'>
                  <input
                    className='artist__input visually-hidden'
                    type='radio'
                    name='answer'
                    value={`artist-${i}`}
                    id={`answer-${i + 1}`}
                    onClick={(evt) => {
                      this.setState({isPlaying: false});
                      onAnswer(evt.target.value);
                    }}
                  />
                  <label className='artist__name' htmlFor={`answer-${i + 1}`}>
                    <img
                      className='artist__picture'
                      src={answer.picture}
                      alt={answer.artist}
                    />
                    {answer.artist}
                  </label>
                </div>
              );
            })}
          </form>
        </section>
      </section>
    );
  }
}

ArtistQuestionScreen.propTypes = {
  screenIndex: PropTypes.number.isRequired,
  onAnswer: PropTypes.func.isRequired,
  question: PropTypes.shape({
    answers: PropTypes.array.isRequired,
    song: PropTypes.shape({
      artist: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};
