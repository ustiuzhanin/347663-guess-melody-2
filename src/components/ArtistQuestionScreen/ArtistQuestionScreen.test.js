import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './ArtistQuestionScreen.jsx';

test(`ArtistQuestionScreen's shapshot`, () => {
  const tree = renderer
    .create(
        <ArtistQuestionScreen
          screenIndex={0}
          onAnswer={jest.fn()}
          question={{answers: [{}]}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
