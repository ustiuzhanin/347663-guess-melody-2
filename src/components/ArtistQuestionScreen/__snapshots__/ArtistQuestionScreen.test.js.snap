// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen's shapshot 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            className="artist__picture"
          />
        </label>
      </div>
    </form>
  </section>
</section>
`;
